File: crates/mithril-oxide-sys/cpp/Dialect/Shape/IR/Shape.cpp
#include "Shape.hpp"

#include <mlir/Dialect/Shape/IR/Shape.h>
#include <mlir/IR/MLIRContext.h>


namespace mithril_oxide_sys::shape {

const void *ShapeType_get(MLIRContext *ctx)
{
    return ShapeType::get(ctx);
}

const void *SizeType_get(MLIRContext *ctx)
{
    return SizeType::get(ctx);
}

const void *ValueShapeType_get(MLIRContext *ctx)
{
    return ValueShapeType::get(ctx);
}

const void *WitnessType_get(MLIRContext *ctx)
{
    return WitnessType::get(ctx);
}

} // namespace mithril_oxide_sys::shape

File: crates/mithril-oxide-sys/src/Dialect/Shape/IR/Shape.rs
pub use self::ffi::DeviceAsyncTokenType_get;

#[cxx::bridge]
mod ffi {
    #[namespace = "mlir"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/Shape/IR/Shape.hpp");

        type MLIRContext = crate::IR::MLIRContext::MLIRContext;
    }

    #[namespace = "mithril_oxide_sys::pdl"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/Shape/IR/Shape.hpp");

        type c_void = crate::IR::Value::ffi::c_void;

        pub unsafe fn ShapeType_get(context: *mut MLIRContext) -> *const c_void;
        pub unsafe fn SizeType_get(context: *mut MLIRContext) -> *const c_void;
        pub unsafe fn ValueShapeType_get(context: *mut MLIRContext) -> *const c_void;
        pub unsafe fn WitnessType_get(context: *mut MLIRContext) -> *const c_void;
    }
}
